src/update-process-env.js

```diff
@@ -34,7 +31,7 @@ function updateProcessEnv (launchEnv) {
     }
 
     for (let key in envToAssign) {
-      if (!ENVIRONMENT_VARIABLES_TO_PRESERVE.has(key)) {
+      if (!ENVIRONMENT_VARIABLES_TO_PRESERVE.has(key) || (!process.env[key] && envToAssign[key])) {
```

Comment:
Why would `envToAssign[key]` be falsy? You're thinking like a blank string environment variable? The key definitely exists because we're looping over keys in the line above.